src/vm/gccover.cpp

```diff
@@ -84,6 +84,10 @@ void SetupAndSprinkleBreakpoints(
     gcCover->callerThread      = 0;
     gcCover->doingEpilogChecks = true;    
 
+    // Expose m_GcCover early before we add breakpoints.
+    _ASSERTE(!pMD->m_GcCover);
+    *EnsureWritablePages(&pMD->m_GcCover) = gcCover;
```

Comment:
Just to make sure I understand the reason for this change - it is to ensure that if another thread was running the function into which we sprinkle the breakpoints and it hits one of the breakpoints, we have the m_GcCover set, right?

Reply:
Yes, a fairly common spurious failure mode is for a thread to hit a GC cover interrupt and then discover that m_GcCover is null. There are comments elsewhere about locks preventing other threads from running the code while it's being instrumented but it does seem to happen.

Reply:
This will replace easy to diagnose failure mode (m_GcCover is null) with a different failure mode that is much harder to diagnose (m_GcCover still being initialized by one thread while the other thread is using it to handle coverage interrupts).

Should we rather fix the race that leads to code running before `m_GcCover` is initialized?

Reply:
It doesn't appear that m_GcCover is lacking anything important for correctness when it's published early. A racing thread may see that `lastMD` was not cleared and so change the logging a little, but I currently don't see how the early publish will lead to a new class of hard to diagnose failures.

But this kind of thing can be hard to reason about with confidence and it would be better to eliminate the races. The one I saw frequently was in regex-redux-5 and came from a dynamically generated regex method. I'll look more closely at that case and try and figure out where the race comes from.